### PR TITLE
fix: Adds compatibility for plugin interface 3.0

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change
(A few sentences about this PR)

## Related issues
- Link to issue1 here
- Link to issue1 here

## Test Plan
(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes
(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates
- [ ] Changelog has been updated
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
   - In `build.gradle`
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR
- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
```
Comment (rishabhpoddar) [inline CHANGELOG.md:8]: 1.24 version

File: src/main/java/io/supertokens/storage/mongodb/Start.java [hunk 2]
```diff
@@ -340,18 +348,18 @@ void handleKillSignalForWhenItHappens() {
...
-    public boolean canBeUsed(String configFilePath) {
-        return Config.canBeUsed(configFilePath);
+    public boolean canBeUsed(JsonObject configJson) {
+        return Config.canBeUsed(configJson);
     }
 
     @Override
-    public boolean isUserIdBeingUsedInNonAuthRecipe(String className, String userId) throws StorageQueryException {
+    public boolean isUserIdBeingUsedInNonAuthRecipe(AppIdentifier appIdentifier, String className, String userId) throws StorageQueryException {
         /* We do not do anything here since Mongodb does not support UserIdMapping */
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/mongodb/Start.java]: throw unsupported operation exception in all such cases

File: src/main/java/io/supertokens/storage/mongodb/Start.java [hunk 3]
```diff
@@ -380,4 +388,41 @@ public boolean setJWTSigningKeyInfoIfNoKeyForAlgorithmExists_Transaction(JWTSign
...
+
+    @Override
+    public String getUserPoolId() {
+        // we do not allow multiple in memory dbs as that is not really useful in any way..
+        return "same-user-pool";
+    }
+
+    @Override
+    public String getConnectionPoolId() {
+        // we do not allow multiple in memory dbs as that is not really useful in any way..
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/mongodb/Start.java]: wrong comment - please search for the word memory

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/storage/mongodb/Start.java (hunk 2 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/mongodb/Start.java]: Done

File: src/main/java/io/supertokens/storage/mongodb/Start.java (hunk 3 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/mongodb/Start.java]: Done

**Review by porcellus — APPROVED** [review]

**Pull request merged**